src/routes/import-request/add.ts

```diff
@@ -0,0 +1,34 @@
+import { ControllerGeneratorOptionsWithClient } from '../../comms/controller';
+import { User } from '../../models/user';
+import { FileToServer, schema as fileToServer } from '../../models/file-to-server';
+import {
+  ImportRequest,
+  importRequestSchema,
+} from '../../models/import-request';
+
+type Request = {
+  body: Required<FileToServer>;
+}
+
+type Response = (ImportRequest) & {
+  createdByUsername: User['name'];
+};
+
+const controllerGeneratorOptions: ControllerGeneratorOptionsWithClient = {
+  method: 'post',
+  path: '/',
+  body: fileToServer
+    .options({ presence: 'required' })
+    .required(),
+  right: { environment: 'IMPORT' },
+  response:
+    importRequestSchema,
```

Comment:
this should have a `.required()` behind it I think?

Reply:
should always be there. indeed.